### PR TITLE
Wrap string

### PR DESCRIPTION
**Pull request by duncanspumpkin**
Fixes https://github.com/OpenLoco/OpenLoco/issues/1557 and a few other issues.

### CODE REVIEW

File: src/OpenLoco/Objects/VehicleObject.cpp [hunk 1]
```diff
@@ -184,14 +184,14 @@ namespace OpenLoco
...
     {
-        if (cost_index > 32)
-        {
-            return false;
-        }
-        if (run_cost_index > 32)
-        {
-            return false;
-        }
+        //if (cost_index > 32)
```
Comment (duncanspumpkin) [inline src/OpenLoco/Objects/VehicleObject.cpp:187]: revert

File: src/OpenLoco/Objects/VehicleObject.cpp [hunk 2]
```diff
@@ -229,10 +229,10 @@ namespace OpenLoco
             return false;
         }
 
-        if (rack_speed > speed)
-        {
-            return false;
-        }
+        //if (rack_speed > speed)
```
Comment (duncanspumpkin) [inline src/OpenLoco/Objects/VehicleObject.cpp:232]: revert

File: src/OpenLoco/Graphics/Gfx.cpp [hunk 3]
```diff
@@ -1338,15 +1335,119 @@ namespace OpenLoco::Gfx
...
-        registers regs;
-        regs.esi = X86Pointer(buffer);
-        regs.di = stringWidth;
-        call(0x00495301, regs);
+        // std::vector<const char*> wrap; TODO: refactor to return pointers to line starts
+        uint16_t wrapCount = 0;
+        auto font = *_currentFontSpriteBase;
+        uint16_t maxWidth = 0;
+
+        for (auto* ptr = buffer; *ptr != '\0';)
```
[the remark this answers is not in the document]
Comment (duncanspumpkin) [reply src/OpenLoco/Graphics/Gfx.cpp:1348]: /0 is already what most people expect for null termination i wouldn't really want to change that

[the remark this answers is not in the document]
Comment (LeftofZen) [reply src/OpenLoco/Graphics/Gfx.cpp:1348]: I don't mean like that, I mean its very easy to mistype it as `'0'` and not realise. It was just a general comment, I'm not asking to introduce a constant for this or anything

File: src/OpenLoco/Graphics/Gfx.cpp [hunk 4]
```diff
@@ -1338,15 +1335,119 @@ namespace OpenLoco::Gfx
...
+                            break;
+                        }
+                        case ControlCodes::font_small:
+                            font = Font::small;
+                            break;
+                        case ControlCodes::font_large:
+                            font = Font::large;
+                            break;
+                        case ControlCodes::font_bold:
+                            font = Font::medium_bold;
```
[the remark this answers is not in the document]
Comment (duncanspumpkin) [reply src/OpenLoco/Graphics/Gfx.cpp:1378]: no only medium has a bold/normal

File: src/OpenLoco/Graphics/Gfx.cpp [hunk 5]
```diff
@@ -1338,15 +1335,119 @@ namespace OpenLoco::Gfx
...
+                }
+                else if (*ptr >= ControlCodes::twoArgBegin && *ptr < ControlCodes::twoArgEnd)
+                {
+                    ptr += 2;
+                }
+                else if (*ptr >= ControlCodes::fourArgBegin && *ptr < ControlCodes::fourArgEnd)
+                {
+                    switch (*ptr)
+                    {
+
```
[the remark this answers is not in the document]
Comment (duncanspumpkin) [reply src/OpenLoco/Graphics/Gfx.cpp]: updated

**Pull request merged**